circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/relations/ecc_vm/ecc_transcript_relation.cpp

```diff
@@ -0,0 +1,263 @@
+     * If q_mul = 1, pc decrements by !z1_zero + !z2_zero, else pc decrements by 0
+     * @note pc starts out at its max value and decrements down to 0. This keeps the degree of the pc polynomial smol
+     */
+    Accumulator pc_delta = pc - pc_shift;
+    std::get<3>(accumulator) +=
+        is_not_first_row * (pc_delta - q_mul * ((-z1_zero + 1) + (-z2_zero + 1))) * scaling_factor;
+
+    /**
+     * @brief Validate `msm_transition` is well-formed.
+     *
+     * If the current row is the last mul instruction in a multiscalar multiplication, msm_transition = 1.
+     * i.e. if q_mul == 1 and q_mul_shift == 0, msm_transition = 1, else is 0
+     */
+    auto msm_transition_check = q_mul * (-q_mul_shift + 1);
+    std::get<4>(accumulator) += (msm_transition - msm_transition_check) * scaling_factor;
+
+    /**
+     * @brief Validate `msm_count` resets when we end a multiscalar multiplication.
+     * msm_count tracks the number of scalar muls in the current active multiscalar multiplication.
+     * (if no msm active, msm_count == 0)
+     * If current row ends an MSM, `msm_count_shift = 0` (msm_count value at next row)
+     */
+    std::get<5>(accumulator) += (msm_transition * msm_count_shift) * scaling_factor;
+
+    /**
+     * @brief Validate `msm_count` updates correctly for mul operations.
+     * msm_count updates by (!z1_zero + !z2_zero) if current op is a mul instruction (and msm is not terminating at next
+     * row).
+     */
+    auto msm_count_delta = msm_count_shift - msm_count; // degree 4
+    std::get<6>(accumulator) += is_not_first_row * (-msm_transition + 1) *
+                                (msm_count_delta - q_mul * ((-z1_zero + 1) + (-z2_zero + 1))) * scaling_factor;
+
+    /**
+     * @brief Add multiscalar multiplication result into transcript accumulator.
+     * If `msm_transition == 1`, we expect msm output to be present on (transcript_msm_x, transcript_msm_y).
+     * (this is enforced via a lookup protocol).
+     * If `is_accumulator_empty == 0`, we ADD msm output into transcript_accumulator.
+     * If `is_accumulator_empty = =1`, we ASSIGN msm output to transcript_accumulator.
+     * @note the output of an msm cannot be point at infinity (will create unsatisfiable constraints in
+     * ecc_msm_relation). We assume this does not affect statistical completeness for honest provers. We should validate
+     * this!
+     */
+    auto add_msm_into_accumulator = msm_transition * (-is_accumulator_empty + 1);
+    auto x3 = transcript_accumulator_x_shift;
+    auto y3 = transcript_accumulator_y_shift;
+    auto x1 = transcript_accumulator_x;
+    auto y1 = transcript_accumulator_y;
+    auto x2 = transcript_msm_x;
+    auto y2 = transcript_msm_y;
+    auto tmpx = (x3 + x2 + x1) * (x2 - x1) * (x2 - x1) - (y2 - y1) * (y2 - y1);
+    auto tmpy = (y3 + y1) * (x2 - x1) - (y2 - y1) * (x1 - x3);
+    std::get<7>(accumulator) += tmpx * add_msm_into_accumulator * scaling_factor; // degree 5
```

Comment:
What is enforcing that transcript_msm!=P1 here? If y2==y1 and x2==x1 both constraints are zero